docs/config-docs.md

```diff
@@ -0,0 +1,374 @@
+
+## Contents
+
+- [How to install Eww](#how-to-install-eww)
+	- [Prerequisites](#prerequisites)
+	- [Building](#building)
+	- [Starting  Eww](#starting-eww)
+-  [Configuration](#configuration)
+	- [Placing the configuration file](#placing-the-configuration-file)
+	- [Variables](#variables)
+		- [The var tag](#the-var-tag)
+		- [The script-var tag](#the-script-var-tag)
+			- [Tail](#tail)
+	- [The Definitions  block](#the-definitions-block)
+		- [Custom Widgets](#custom-widgets)
+	- [The Windows block](#the-windows-block)
+    - [Widgets made in Eww](#widgets-made-in-eww)
+- [GTK](#gtk)
+	- [GTK-Temeing / Eww Themeing](#gtk-themeing)
+	- [GTK-Debugger](#gtk-debugger)
+- [Troubleshooting](#troubleshooting)
+	- [My scss isn't being loaded](#my-scss-isnt-being-loaded)
+	- [Eww can't find my configuration file](#eww-cant-find-my-configuration-file)
+	- [Something isn't styled correctly!](#something-isnt-styled-correctly)
+
+
+## How to install Eww 
+
+### Prerequisites
+
+* rustc
+* cargo (nightly toolchain)
+
+Rather than with your system package manager, I recommend installing it using  [rustup](https://rustup.rs/), as this makes it easy to use the nightly toolchain, which is necessary to build eww.
+
+### Building
+
+Once you have the Prerequisites ready, you're ready to install and build eww.
+
+First clone the repo:
+```bash
+git clone https://github.com/elkowar/eww
+```
+then enter it.
+```bash
+cd eww
+```
+and then to build:
+```bash
+cargo build --release
+```
+
+### Running eww
+Once you've built it you can now run it by entering:
+```bash
+cd target/release
+```
+and then make the Eww binary executable
+```bash
+chmod +x ./eww
+```
+and then to run it do
+```
+./eww open <window_name>
+```
+`<window_name>` is the name of the window, see [The windows block](#the-windows-block).
+
+## Configuration
+
+For specific built in widgets `<box>, <text>, <slider>, etc` see [Widget Documetation]()
+
+### Placing the configuration file
+
+The configuration file and the scss file should lay in your `$HOME/.config/eww` folder. The xml file should be named `eww.xml` and the scss should be named `eww.scss`
+So the directory structure should look like this:
+```
+$HOME
+└──.config
+   └──eww
+	  ├──eww.xml
+	  └──eww.scss
+```
+
+### Variables
+
+If you create a `<var>` or a `<script-var>`, you can reference them in your `<box>` by doing `{{var}}`. Where `var` is your variable name.
+
+#### The `<var>` tag
+Allows you to repeat the same text multiple times through  without retyping it multiple times.
+
+Example:
+```xml
+<variables>
+	<var name="banana">I like bananas.</var>
+</variables>
+```
+
+and then reference it in your `<box>` by doing:
+```xml
+<box>
+	{{banana}}
+</box>
+```
+ the `banana` is the variable name we set by doing: 
+ ```xml
+ <var name="banana">
+```
+ If we were to set it to:
+ ```xml
+<var name="potatoes">
+```
+We would reference it by doing:
+```xml
+<box>
+	{{potatoes}}
+</box>
+```
```

Comment:
```suggestion
Example: This will define a variable named `banana`, with the default value "I like bananas.".
```xml
<variables>
	<var name="banana">I like bananas.</var>
</variables>
```

you can then reference it in your widgets by doing:

```xml
<box>
	{{banana}}
</box>
```

To change the value of the variable, and thus change the UI, you can run `eww update banana "I like apples"`

```

Can't get this suggestion to be formatted correctly, i hope you get what i mean. The section doesn't have to be thiiiiissss long :D